### PR TITLE
Ex 6 - React

### PR DESCRIPTION
**Pull request by liortal2**
I'm new to React, first time using it, took some time to get the hang of it but eventually got it to work. 
Please let me know if I it is better to split to more components or less?
Thanks !!

### CODE REVIEW

File: client/src/components/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      </nav>
+   );
+}
+
+function CustomLink({ to, children }) {
+   const resolvedPath = useResolvedPath(to);
+   const isActive = useMatch({ path: resolvedPath.pathname, end: true });
+
+   return (
+      <li className={`${isActive ? navbarCSS.active : ""}`}>
```
Comment (IdanDavidi) [inline client/src/components/Navbar.jsx]: you can use `classnames`
https://www.npmjs.com/package/classnames

```suggestion
      <li className={classnames({ [navbarCSS.active]: isActive })}>
```

File: client/src/components/Button.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import PropTypes from "prop-types";
+
+function Button({ buttonHandler, className, innerText }) {
```
Comment (IdanDavidi) [inline client/src/components/Button.jsx]: it make more sense (and it's a standard) to call it onClick
```suggestion
function Button({ onClick, className, innerText }) {
```

File: client/src/components/HomeCard.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   }, []);
+
+   return (
+      <div className={homeCardCSS.card}>
+         <Titles />
+         <AddBar setTasks={setTasks} setIsLoading={setIsLoading} />
+         <SortButtons tasks={tasks} setTasks={setTasks} />
+         <div className={homeCardCSS["tasks-and-clear-container"]}>
+            <TaskList tasks={tasks} setTasks={setTasks} setIsLoading={setIsLoading} />
+            {isLoading === true ? <Loader /> : ""}
```
Comment (IdanDavidi) [inline client/src/components/HomeCard.jsx]:
```suggestion
            {isLoading ? <Loader /> : ""}
```

or

```suggestion
            {isLoading && <Loader />}
```

File: client/src/components/TaskList.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+         } catch (error) {
+            console.error(error);
+         }
+         setIsLoading(false);
+      })();
+   }, [setTasks, setIsLoading]);
+
+   const tasksList = useMemo(
+      () =>
+         [...tasks].map((task) => (
```
Comment (IdanDavidi) [inline client/src/components/TaskList.jsx]: you don't really have to use spread operator here - it creates redundant extra array
```suggestion
         tasks.map((task) => (
```

**Review by IdanDavidi — APPROVED** [review]
@liortlv great job!! can you please attach a video that demonstrate the application features? (I wasn't able to run it locally easily because I need to setup a DB for that)

File: client/src/components/Navbar.jsx (hunk 1 above)
Comment (liortal2) [reply client/src/components/Navbar.jsx]: Oh cool, just added it here and in another comopnent,  thanks!

**Pull request merged**